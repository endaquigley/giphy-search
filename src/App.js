import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import Gallery from './Gallery';
import Pagination from './Pagination';
import * as actions from './actions';

const Header = styled.header`
  padding: 20px;
  text-align: center;
`;

const Content = styled.main`
  background-color: var(--secondary-colour);
`;

const Footer = styled.footer`
  color: white;
  padding: 20px;
`;

const Heading = styled.h1`
  font-size: 50px;
  line-height: 1;
  margin-top: 0;
  margin-bottom: 15px;
`;

class App extends Component {

  render() {
    return (
      <Fragment>

        <Header>
          <Heading>
            { this.props.query === 'dog' ? 'üê∂' : 'üê±' }
          </Heading>

          <button onClick={ () => this.props.updateQuery('dog') } disabled={ this.props.query === 'dog' }>Dog Search</button>
          <button onClick={ () => this.props.updateQuery('cat') } disabled={ this.props.query === 'cat' }>Cat Search</button>
        </Header>

        <Content>
          <Gallery />
        </Content>

        <Footer>
          <Pagination
            page={ this.props.page }
            prevPage={ () => this.props.prevPage() }
            nextPage={ () => this.props.nextPage() }
          />
        </Footer>

      </Fragment>
    );
  }

};

const mapStateToProps = (state) => {
  return {
    page: state.page,
    query: state.query
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    prevPage: () => dispatch(actions.prevPage()),
    nextPage: () => dispatch(actions.nextPage()),
    updateQuery: (query) => {
      return dispatch(actions.updateQuery(query))
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
