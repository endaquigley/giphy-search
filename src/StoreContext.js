import { createContext } from "react";

const StoreContext = createContext();
const { Provider } = StoreContext;

export { Provider };
export default StoreContext;
