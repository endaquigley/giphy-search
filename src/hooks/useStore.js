import { useReducer } from "react";
import { reducer, initialState } from "../reducer";

const useStore = () => {
  return useReducer(reducer, initialState);
};

export default useStore;
