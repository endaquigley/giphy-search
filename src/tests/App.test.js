import React from "react";
import { render } from "react-testing-library";

import App from "../App";
import useStore from "../hooks/useStore";
import { Provider } from "../StoreContext";

import "react-testing-library/cleanup-after-each";

const Store = React.memo(({ children }) => {
  return children(useStore());
});

describe("App Component", () => {
  it("renders without crashing", () => {
    const { container } = render(
      <Store>
        {([state, dispatch]) => (
          <Provider value={{ state, dispatch }}>
            <App />
          </Provider>
        )}
      </Store>
    );

    expect(container).toMatchSnapshot();
  });
});
