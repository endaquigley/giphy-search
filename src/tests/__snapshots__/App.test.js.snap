// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<div>
  <header
    class="sc-gZMcBi iCyDGX"
  >
    <h1
      class="sc-bdVaJa hMOmBt"
    >
      üê∂
    </h1>
    <button
      class="sc-bwzfXH bwQeco"
      disabled=""
    >
      Dog Search
    </button>
    <button
      class="sc-bwzfXH bwQeco"
    >
      Cat Search
    </button>
  </header>
  <main
    class="sc-gqjmRU dwogNk"
  >
    <div
      class="sc-htoDjs bxEQbg"
    />
  </main>
  <footer
    class="sc-VigVT kUMmUu"
  >
    <div
      class="sc-dnqmqq hKHdbA"
    >
      <button
        class="sc-iwsKbI bmZaqg"
        disabled=""
      >
        Prev Page
      </button>
      <span>
        Page 
        1
      </span>
      <button
        class="sc-iwsKbI bmZaqg"
      >
        Next Page
      </button>
    </div>
  </footer>
</div>
`;
